src/pages/index.js

```diff
@@ -3,35 +3,117 @@ import {
 import FormValidator from "../components/FormValidator.js";
 import UserInfo from "../components/UserInfo.js";
 import Card from "../components/Card.js";
 import Section from "../components/Section.js";
+import Api from "../components/Api.js";
 
 const pageUserInfo = new UserInfo({
   nameSelector: ".profile__title",
   jobSelector: ".profile__description",
+  avatarSelector: "#profile-avatar",
+});
+
+const api = new Api({
+  baseURL: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "98cf51ba-481b-4eec-9343-23efbe8fbf82",
+    "Content-Type": "application/json",
+  },
+});
+
+let user = api.getUserInfo();
+let apiCards = api.getInitialCards();
+let cardSection;
+
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    (cardData) => {
+      imageViewerPopup.open(cardData);
+    },
+    (event) => {
+      deleteCardPopup.open();
+      deleteCardPopup.setSubmitAction(() => {
+        deleteCardPopup.setLoadingState(true, "Yes");
+        api.deleteCard(cardData._id).then(() => {
+          deleteCardPopup.setLoadingState(false, "Yes");
+          card.remove();
+        });
+      });
+    },
+    (cardId) => {
+      console.log("card isLiked value starts as", card.getLikes());
+      let liked = card.getLikes();
+      if (!liked) {
+        api.addLike(cardId).then((data) => {
+          console.log("card isLiked value is now", data.isLiked);
+        });
+      } else {
+        api.deleteLike(cardId).then((data) => {
+          console.log("card isLiked value is now", data.isLiked);
+        });
+      }
+    }
+  );
+  return card.getView();
+}
+
+Promise.all([user, apiCards]).then(([userData, initialCards]) => {
+  user = userData._id;
+  // console.log("userData", userData);
+  pageUserInfo.setUserInfo(userData);
+  pageUserInfo.setUserAvatar(userData);
+  console.log(initialCards);
+
+  cardSection = new Section(
+    {
+      items: initialCards,
+      renderer: (initialCardData) => {
+        const card = createCard(initialCardData);
+        cardSection.prependItem(card);
+      },
+    },
+    ".cards__list"
+  );
+
+  cardSection.renderItems();
+  // pageUserInfo.setUserAvatar(userData.avatar)
 });
 
 function fillProfileForm(e) {
-  const currentInfo = pageUserInfo.getUserInfo();
+  let currentInfo = pageUserInfo.getUserInfo();
   profileTitleInput.value = currentInfo.name;
   profileDescriptionInput.value = currentInfo.job;
 }
 
 const profileEditForm = new PopupWithForm(
   "#profile-edit-popup",
   (inputValues) => {
-    pageUserInfo.setUserInfo(inputValues);
-    profileEditForm.close();
+    //pageUserInfo.setUserInfo(inputValues);
+    profileEditForm.setLoadingState(true);
+    api.updateUserInfo(inputValues).then(() => {
+      api
+        .getUserInfo()
+        .then((data) => pageUserInfo.setUserInfo(data))
+        .finally(() => {
+          profileEditForm.setLoadingState(false);
+          profileEditForm.close();
```

Comment:
Needs correction: Сlose modal in `then` block, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again.

Reply:
corrected